src/logger.service.ts

```diff
@@ -27,77 +25,101 @@ const Levels = [
+      serverLogLevel: NgxLoggerLevel.OFF
+    };
     this._serverLogLevel = this.options.serverLogLevel;
     this._clientLogLevel = this.options.level;
-    this._isIE = isPlatformBrowser(platformId) &&
-      !!(navigator.userAgent.indexOf('MSIE') !== -1 || navigator.userAgent.match(/Trident\//) || navigator.userAgent.match(/Edge\//));
+    this._isIE = isPlatformBrowser(this.platformId) && !!(navigator.userAgent.indexOf('MSIE') !== -1
+      || navigator.userAgent.match(/Trident\//)
+      || navigator.userAgent.match(/Edge\//));
   }
 
-  trace(message, ...additional: any[]) {
+  public trace(message, ...additional: any[]): void {
     this._log(NgxLoggerLevel.TRACE, true, message, additional);
   }
 
-  debug(message, ...additional: any[]) {
+  public debug(message, ...additional: any[]): void {
     this._log(NgxLoggerLevel.DEBUG, true, message, additional);
   }
 
-  info(message, ...additional: any[]) {
+  public info(message, ...additional: any[]): void {
     this._log(NgxLoggerLevel.INFO, true, message, additional);
   }
 
-  log(message, ...additional: any[]) {
+  public log(message, ...additional: any[]): void {
     this._log(NgxLoggerLevel.LOG, true, message, additional);
   }
 
-  warn(message, ...additional: any[]) {
+  public warn(message, ...additional: any[]): void {
     this._log(NgxLoggerLevel.WARN, true, message, additional);
   }
 
-  error(message, ...additional: any[]) {
+  public error(message, ...additional: any[]): void {
     this._log(NgxLoggerLevel.ERROR, true, message, additional);
   }
 
-  private _timestamp() {
-    return new Date().toISOString();
+  private _timestamp(): string {
+    return new Date()
+      .toISOString();
   }
 
-  private _logOnServer(level: NgxLoggerLevel, message, additional: any[]) {
-    if (!this.options.serverLoggingUrl) {
+  private _logOnServer(level: NgxLoggerLevel, message, additional: any[] = []): void {
+    // If the loggingUrl is not set or if the user provides a serverLogLevel and the current level is than that, do not log.
+    if (!this.options.serverLoggingUrl || level < this._serverLogLevel) {
       return;
     }
 
-    // if the user provides a serverLogLevel and the current level is than that do not log
-    if (level < this._serverLogLevel) {
-      return;
+    const headers = new HttpHeaders()
+      .set('Content-Type', 'application/json');
+
+    let messageToLog = '';
+
+    try {
+      messageToLog = JSON.stringify(message);
+    } catch (e) {
+      messageToLog = 'The provided "message" value could not be parsed with JSON.stringify().';
     }
 
-    const headers = new HttpHeaders().set('Content-Type', 'application/json');
 
-    this.http.post(this.options.serverLoggingUrl, {
-      level: Levels[level],
-      message: message,
-      additional: additional,
-      timestamp: this._timestamp()
-    }, {headers})
+    const additionalToLog: Array<string | null | undefined> = additional === null || additional === undefined ? []
```

Comment:
Please try to simplify this. Although it works, multiple levels of ternary operators with the map and the try/catch, it can be difficult to easily see what is being accomplished.